pkg/controller/vsphere/reconciler_test.go

```diff
@@ -2220,8 +2240,13 @@ func TestReconcileMachineWithCloudState(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	createTagAndCategory(session, zoneKey, testZone)
-	createTagAndCategory(session, regionKey, testRegion)
+	if err := createTagAndCategory(session, zoneKey, testZone); err != nil {
+		t.Fatalf("cannot create tag and category: %v", err)
+	}
+
+	if err := createTagAndCategory(session, regionKey, testRegion); err != nil {
```

Comment:
I'm thinking about introducing gomega here.

P.S. Just thought, lets address this in another pr